src/collection/mod.rs

```diff
@@ -766,6 +766,14 @@ impl<'a, C: ClientExt> Collection<C> {
             deserialize_response(self.session.request(req).await?.body())?;
         Ok(resp)
     }
+
+    /// Returns a new Collection with its `session` updated with the transaction id
+    pub fn copy_with_transaction(&self, transaction_id: String) -> Result<Self, ClientError> {
```

Comment:
I think here is a Clone semantic rather than Copy. Maybe `clone_with_transaction` is better?